extension/test/data_test.ts

```diff
@@ -69,4 +69,166 @@ describe('data.ts', function () {
+          { entry: 'ア /(int) (1) ah/oh/(int) (2) hey!/', reason: undefined },
+        ],
+        hasMore: false,
+        title: '',
+        index: 0,
+        matchLen: 1,
+      };
+
+      it('when max is 1, returns 1 data entry as text', function () {
+        const singleDataEntryAsText = rcxDict.makeText(
+          wordDictEntry,
+          /* max= */ 1
+        );
+
+        expect(singleDataEntryAsText).to.equal(
+          'あ\t(int) (1) ah; oh; (int) (2) hey!\n'
+        );
+      });
+
+      it('when max is 2, returns 2 data entries as text', function () {
+        const twoDataEntriesAsText = rcxDict.makeText(
+          wordDictEntry,
+          /* max= */ 2
+        );
+
+        expect(twoDataEntriesAsText).to.equal(
+          'あ\t(int) (1) ah; oh; (int) (2) hey!\nア\t(int) (1) ah; oh; (int) (2) hey!\n'
+        );
+      });
+
+      it('when max is greater than the number of data entries, returns all data entries as text', function () {
+        const allDataEntriesAsText = rcxDict.makeText(
+          wordDictEntry,
+          /* max= */ 1000
+        );
+
+        expect(allDataEntriesAsText).to.equal(
+          'あ\t(int) (1) ah; oh; (int) (2) hey!\nア\t(int) (1) ah; oh; (int) (2) hey!\n'
+        );
+      });
+    });
+
+    describe('with a name dict entry', function () {
+      // nameDictEntry was generated with: rcxDict.wordSearch('あ', /* doNames= */ true)
+      const nameDictEntry = {
+        kanji: '',
+        onkun: '',
+        nanori: '',
+        bushumei: '',
+        misc: {},
+        eigo: '',
+        hasNames: true,
+        data: [
+          { entry: '亜 [あ] /(f) A/', reason: undefined },
+          { entry: '阿 [あ] /(s) A/', reason: undefined },
+        ],
+        hasMore: false,
+        title: '',
+        index: 0,
+        matchLen: 1,
+      };
+
+      it('when max is 1, returns 1 data entry as text', function () {
+        const singleDataEntryAsText = rcxDict.makeText(
+          nameDictEntry,
+          /* max= */ 1
+        );
+
+        expect(singleDataEntryAsText).to.equal('亜\tあ\t(f) A\n');
+      });
+
+      it('when max is 2, returns 2 data entries as text', function () {
+        const twoDataEntriesAsText = rcxDict.makeText(
+          nameDictEntry,
+          /* max= */ 2
+        );
+
+        expect(twoDataEntriesAsText).to.equal('亜\tあ\t(f) A\n阿\tあ\t(s) A\n');
+      });
+
+      it('when max is greater than the number of data entries, returns all data entries as text', function () {
+        const allDataEntriesAsText = rcxDict.makeText(
+          nameDictEntry,
+          /* max= */ 1000
+        );
+
+        expect(allDataEntriesAsText).to.equal('亜\tあ\t(f) A\n阿\tあ\t(s) A\n');
+      });
+    });
+
+    describe('with a kanji dict entry', function () {
+      // kanjiDictEntry was generated with: rcxDict.kanjiSearch('両')
+      const kanjiDictEntry = {
+        kanji: '両',
+        onkun: 'リョウ、 てる、 ふたつ',
+        nanori: 'もろ',
+        bushumei: '',
+        misc: {
```

Comment:
For new code, I would probably have a single test for each misc field but for a regression test it's not required.